src/analysis/processing/pdal/qgspdalalgorithmbase.cpp

```diff
@@ -288,4 +289,37 @@ QVariantMap QgsPdalAlgorithmBase::processAlgorithm( const QVariantMap &parameter
   return outputs;
 }
 
+QgsPointCloudLayer *QgsPdalAlgorithmBase::parameterAsPointCloudLayer( const QVariantMap &parameters, const QString &name, QgsProcessingContext &context, QgsProcessing::LayerOptionsFlags flags ) const
+{
+  QgsPointCloudLayer *layer = QgsProcessingParameters::parameterAsPointCloudLayer( parameterDefinition( name ), parameters, context, flags );
+
+  if ( !layer )
+    return nullptr;
+
+  if ( layer->dataProvider()->name() == QStringLiteral( "copc" ) )
+  {
+    return layer;
+  }
+
+  const QString copcFileName = QgsPdalAlgorithmBase::copcIndexFile( layer->source() );
+  const QFileInfo fiCopcFile( copcFileName );
+
+  if ( fiCopcFile.exists() )
```

Comment:
micro-optimization: there is QFileInfo::exists( filename ) static function, which should be faster